server.py

```diff
@@ -113,5 +129,17 @@ def get_verified_email(jwt):
     return {'email': payload['sub']}
 
 
-host = 'localhost' if 'localhost' in META['RP_ORIGIN'] else '0.0.0.0'
-run(server='gunicorn', host=host, port=int(getenv('PORT', '8080')))
+HANDLERS = {'index': index, 'login': login, 'static': static}
+STATUS = {200: '200 OK', 400: '400 Bad Request'}
+
+def application(env, respond):
+    pi = [] if not env['PATH_INFO'] else env['PATH_INFO'].strip('/').split('/')
+    handler = pi[0] if pi and pi[0] else 'index'
+    status, headers, content = HANDLERS[handler](env)
+    respond(STATUS[status], list(headers.items()))
+    return [content]
+
+
+if __name__ == '__main__':
+    host, port = parse.urlparse(META['RP_ORIGIN']).netloc.split(':')
+    simple_server.make_server(host, int(port), application).serve_forever()
```

Comment:
I think it is single-threaded, yes. In my opinion, for the purposes of being a demo RP, all of that does not matter; this code exists solely to prove out that it works. A real RP will be embedded in a larger app anyway. As a demo RP, having the lowest possible barrier to access is better, so relying on the stdlib is better than relying on gunicorn. I guess asyncio is an option, but personally I would like this to also work on Python 2.7 out of the box (lowest possible barrier...).